module2/projects/peer_code_share.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: page
+title: Relational Rails - Peer Code Share
+---
+
+_[Back to Relational Rails Home](./relational_rails)_
+
+## Peer Code Share
+
+You will be partnered with another student to review each other's code and provide feedback.
+
+First, exchange Github repository links with the other student, and clone the repository. Set up the application locally by first checking that *their* database is named differently than *yours* - if you `drop` or re-seed a database of the exact same name as your project's, then yours will be affected too! To accomplish this, you could simply add the name of the student at the start of the database name, for example: `brian_relational_rails_development`, `brian_relational_rails_test`. You can find the database configuration in `config/database.yml`. 
+
+Then, in your Terminal, run the following: 
+```ruby
+bundle install
+rails db:{drop,create,migrate,seed}
+rails s
+```
+
+Then, individually, take about 25 minutes to review the other person's code. You should write down answers to the following questions:
+
+1. Can you launch/use the application? Can the user stories be completed accurately? Do you find any bugs as you navigate through any pages? 
+1. Run their overall test suite (`bundle exec rspec`). Are there any failing tests? Are their tests strong/robust? (Strong tests utilize both obvious outcomes and edge cases.)
+1. Check their application's Models and the relationship to each other. Does the one-to-many relationship work? Are there any instance methods in their models? Are their models tested sufficiently (100%)? 
```

Comment:
Do we want them to just look at instance methods?

Reply:
I guess what I was getting at here was, if they have instance methods and they are tested (and those tests pass), then that's sufficient. Maybe in addition, check the model tests to be sure they are strong enough - adding more instances to the scenarios they already tested, for example, is something I ended up telling a lot of students in the past. 

To make this a little more effective, maybe it should also say "Are there any operations in a Controller or a View that should belong in this Model as methods instead?"